go-controller/pkg/node/default_node_network_controller.go

```diff
@@ -136,13 +136,15 @@ func NewDefaultNodeNetworkController(cnnci *CommonNodeNetworkControllerInfo) (*D
 		}
 	}
 
-	nc.apbExternalRouteNodeController, err = apbroute.NewExternalNodeController(
-		cnnci.apbExternalRouteClient,
-		nc.watchFactory.PodCoreInformer(),
-		nc.watchFactory.NamespaceInformer(),
-		stopChan)
-	if err != nil {
-		return nil, err
+	if config.OVNKubernetesFeature.EnableMultiExternalGateway {
+		nc.apbExternalRouteNodeController, err = apbroute.NewExternalNodeController(
+			cnnci.apbExternalRouteClient,
+			nc.watchFactory.PodCoreInformer(),
+			nc.watchFactory.NamespaceInformer(),
+			stopChan)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
nit: maybe this can be initialized closer to which is used as `nc.apbExternalRouteNodeController.Run` so we just check `EnableMultiExternalGateway` once but this is just a thought I had, no problems if it is left as is.

Reply:
I moved the controller initialization to where `nc.apbExternalRouteNodeController.Run` is invoked. But for the master code in `default_network_controller.go`, I keep the `apbroutecontroller.NewExternalMasterController`, since there are many other codes depends on the objects in the controller instance.

Reply:
@pliurh can you explain what happens in those situations where the code depends on `apbroutecontroller` but EnableMultiExternalGateway is not set?

For example:

- `addGWRoutesForPod` seems to be doing something when it might not have to
- Also `apbExternalRouteController.Repair` might be doing something different that what it would be supposed to do, if for example, we restart from a situation where `EnableMultiExternalGateway` was set to a being unset.

Reply:
When `EnableMultiExternalGateway` is unset. The `apbExternalRouteController` will not be running. So all the informers in the controller will not be started. I suppose `addGWRoutesForPod` will do nothing for the apbRoutePolicies.

`apbExternalRouteController.Repair` is more complicated, since it will also do things based on the annotation of pods and namespaces. If a cluster is always with `EnableMultiExternalGateway` flag unchanged, it's safe. However with the case you talked about, if the associated annotations for pods and namespace are not cleaned, I am not sure what will happen.

Maybe @jordigilh can shed some light on this.

Reply:
Please note that `func (oc *DefaultNetworkController) addGWRoutesForPod` still looks like it does something. 
Is the plan to keep legacy egress gw functionality operating through pod & namespace annotations even if this flag is not set?

In regards to the repair, we decided we should not really be concerned with cleaning up if the flag transitioned from enabled to disabled, which is comparable to what we do with other toggle features. However, it looks like the repair might still be creating things even if the flag is unset, from `processOVNRoute` called in Repair.

So should we just consider that the flag also applies to the legacy egress gw support and just check for this flag in all these places and just disable everything?